modules/groovy-dsl/Assertions.md

```diff
@@ -0,0 +1,232 @@
+Checks whether the body of the HTTP response contains a string or something matching a regular expression.
+
+**Parameters:** A string to look for, and optionally parameters that indicate whether the string should be 
+used as a regular expression (```useRegexp```)and whether the search should ignore case (```ignoreCase```).
+
+
+**Examples:** 
+```groovy
+ get 'https://staging-server/customers/1', {
+     asserting {
+         /* matches Smith, SMITH etc */
+         responseContains 'smith', ignoreCase: true
+     }
+ }
+ get 'https://staging-server/customers/1', {
+      asserting {
+          /* matches Smith, SMITH etc, but also any of those strings with an s appended to it */
+          responseContains 'smith(s)?', ignoreCase: true, useRegexp: true
+      }
+  }
+ ```
+The first assertion will accept Smith, SMITH etc. The second one will use "smith(s)" as a regular expression, so it will
+also accept Smiths and SMITHS, for instance.
+
+### responseDoesNotContain
+
+This is simply the inverse of ```responseContains``` (see above). It will assert that the response content does **not**
+contain a string or match a regular expression.
+
+**Parameters:** A string to look for, and optionally key-value pairs that indicate whether the string should be 
+used as a regular expression (```useRegexp```)and whether the search should ignore case (```ignoreCase```).
+
+
+**Examples:** 
+```groovy
+ get 'https://staging-server/customers/1', {
+     asserting {
+         /* matches content not containing the text "smith" */
+         responseDoesNotContain 'smith', ignoreCase: false
+     }
+ }
+ get 'https://staging-server/customers/1', {
+      asserting {
+          /* will fail for all content that contains either "smith" or "smiths" */
+          responseContains 'smith(s)?', ignoreCase: false, useRegexp: true
+      }
+  }
+ ```
+The first assertion will accept Smith, SMITH etc. The second one will use "smith(s)" as a regular expression and will thus
+also accept e.g. Smiths and SMITHS.
+
+### maxResponseTime
+
+Verifies that the response time is lower than the specified max time.
+
+**Parameter:** A time expression in the format ```<number> [millisecond[s])|ms|second[s])|minute[s])]```. If the time unit is omitted,
+milliseconds will be assumed.
+   Here are some valid time expressions:
+   + ```1 second```
+   + ```1.5 minutes```
+   + ```50 milliseconds```
+   + ```50 ms```
+   + ```50```
+Note that the last two expressions are equivalent.
+
+**Examples:** 
+```groovy
+ get 'https://staging-server/customers/1', {
+     asserting {
+         maxResponseTime 2 seconds
+     }
+ }
+ post 'https://staging-server/customers/1', {
+      asserting {
+          maxResponseTime 500 milliseconds
+      }
+  }
+ ```
+ 
+### script
+ 
+Uses a Groovy script to perform an assertion using the current test state. This is an advanced assertion that 
+assumes that the user is familiar with the SoapUI object model. For information about how Groovy script assertions
```

Comment:
link to https://www.soapui.org/scripting---properties/the-soapui-object-model.html ?

Reply:
Good idea